RUFAS/simulation_engine.py

```diff
@@ -132,7 +132,9 @@ def _daily_simulation(self) -> None:
         self.animal_manager.daily_updates(self.feed, self.weather, self.time)
         all_pen_manure_data = self.animal_manager.collect_pen_manure_data()
         self.manure_manager.daily_update(all_pen_manure_data, self.animal_manager.simulation_day)
-        self.field_manager.daily_update_routine(self.weather, self.time)
+        harvested_crops = self.field_manager.daily_update_routine(self.weather, self.time)
+        for crop in harvested_crops:
```

Comment:
```suggestion
        for harvested_crop in harvested_crops:
```
¯\_(ツ)_/¯ maybe doesn't make a huge difference but why not.

Reply:
Also, it's just one line but I was talking to Pooya today about my disentanglement PR and he mentioned the level of abstraction wasn't consistent when I started iterating through all the fields in field_manager and I wondered if this too should maybe be done with this for loop. Maybe a process_harvested_crops() function type thing.

Reply:
We should discuss this more, I would like to flush out what the inconsistency is in the levels of abstraction.

Reply:
Following up: the suggested code is in `SimulationEngine`.

For the second part, we discussed and my cents is that none of the attributes in any of the managers (`FieldManager`, `FeedManager`, etc.) should be interacted with directly. Only the interface of the managers should be engaged with in `SimulationEngine`.